frame/recovery/src/benchmarking.rs

```diff
@@ -0,0 +1,395 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use crate::Pallet;
+use frame_benchmarking::{account, benchmarks, whitelisted_caller};
+use frame_support::traits::{Currency, Get};
+use frame_system::RawOrigin;
+use sp_runtime::traits::{BlockNumberProvider, Bounded};
+
+const SEED: u32 = 0;
+
+#[allow(dead_code)]
+fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
+	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
+}
+
+fn block_number<T: Config>() -> T::BlockNumber {
+	frame_system::Pallet::<T>::current_block_number()
+}
+
+fn get_total_deposit<T: Config>(
+	bounded_friends: &FriendsOf<T>,
+) -> Option<<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance>
+{
+	let friend_deposit = T::FriendDepositFactor::get()
+		.checked_mul(&bounded_friends.len().saturated_into())
+		.unwrap();
+
+	T::ConfigDepositBase::get().checked_add(&friend_deposit)
+}
+
+fn generate_friends<T: Config>(num: u32) -> Vec<<T as frame_system::Config>::AccountId> {
+	// Create friends
+	let mut friends = vec![];
+	for i in 0..num {
+		friends.push(account("friend", i, SEED));
+	}
+	// Sort
+	friends.sort();
+
+	for friend in 0..friends.len() {
+		// Top up accounts of friends
+		T::Currency::make_free_balance_be(
+			&friends.get(friend).unwrap(),
+			BalanceOf::<T>::max_value(),
+		);
+	}
+
+	friends
+}
+
+fn add_caller_and_generate_friends<T: Config>(
```

Comment:
This function is mostly the same as the one above.

Reply:
Alright, I will work on this